.circleci/config.yml

```diff
@@ -23,23 +23,45 @@ jobs:
       - run:
           name: Install dependencies
           command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r .circleci/requirements.txt
+            sudo pip install -r .circleci/requirements.txt
 
       - save_cache:
-          paths:
-            - ./venv
           key: v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
+          paths:
+            - ~/.cache 
 
       - run:
-          name: Linting
+          name: Run linter
           command: |
-            . venv/bin/activate
-            flake8 --max-line-length 160 --per-file-ignores='datascroller/__init__.py:F401' datascroller/
+            # stop the build if there are Python syntax errors or undefined names
+            flake8 datascroller --count --select=E901,E999,F821,F822,F823 \
+              --show-source --statistics
+            # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide      
+            flake8 datascroller --count --exit-zero --max-complexity=10 \
+              --max-line-length=127 --statistics
 
       - run:
-          name: Testing
+          name: Run Tests
           command: |
-            . venv/bin/activate
             pytest test/
+
+      - run:
+          name: init .pypirc
+          command: |
+            echo -e "[pypi]" >> ~/.pypirc
+            echo -e "username = $PYPI_USERNAME" >> ~/.pypirc
+            echo -e "password = $PYPI_PASSWORD" >> ~/.pypirc
+
+      - run:
+          name: create packages
```

Comment:
so we create packages for feature branches? Why? Also, if two feature branches are pushed with the same version number, does this step fail?

Reply:
I believe feature branches are checked out independently of each other, and since they're not being deployed to PyPi, there's no change to fail. I'm almost certain about that given what I know about the checkout process.

Next, should we create packages for feature branches? I wondered about that too. My thought is that it tests the whole process out right up to the deploy without actually going through the final step. But there's not really a situation I can think of where one of these packaging steps would fail because of pushed code. Think we should combine these steps with deploy?

Reply:
No I like them separate but like I said I’m not sure why would they would ever fail. But who knows? Maybe they will at some point and we’ll learn something